api/app/GraphQL/Mutations/DownloadApplicationDoc.php

```diff
@@ -18,9 +19,19 @@ public function __invoke(null $_, array $args)
         $user = Auth::user();
         throw_unless(is_string($user?->id), UnauthorizedException::class);
 
+        $targetApplicant = PoolCandidate::find($args['id'])->load(['user']);
+        $firstName = $targetApplicant?->user?->first_name;
+        $lastName = $targetApplicant?->user?->last_name;
+        if (isset($firstName)) {
+            $firstName = trim(filter_var($firstName, FILTER_SANITIZE_EMAIL));
+        }
+        if (isset($lastName)) {
+            $lastName = trim(filter_var($lastName, FILTER_SANITIZE_EMAIL));
+        }
+
```

Comment:
I don't think `FILTER_SANITIZE_EMAIL` is sufficient for file names. Here is a [list of potentially illegal characters.](https://www.mtu.edu/umc/services/websites/writing/characters-avoid/)

The filter being used here [allows characters in that list](https://www.php.net/manual/en/filter.filters.sanitize.php#:~:text=Remove%20all%20characters%20except%20letters%2C%20digits%20and%20!%23%24%25%26%27*%2B%2D%3D%3F%5E_%60%7B%7C%7D~%40.%5B%5D.).

Reply:
Huh, okay, how about https://github.com/GCTC-NTGC/gc-digital-talent/pull/11475/commits/196836af043d34227e9f1dee9e4222d82605909c

![image](https://github.com/user-attachments/assets/6e7c0435-08c3-4601-aac4-742d56b758ba)

![image](https://github.com/user-attachments/assets/fad6a7a0-de70-4305-bea1-2eecb5371112)

Reply:
So, I know the link I sent said you shouldn't use spaces and while that is true, it was requested to put spaces in the file name :sweat_smile: So, we should keep those even if it is not best practice to do so

Reply:
😆 
Okay, back in they go https://github.com/GCTC-NTGC/gc-digital-talent/pull/11475/commits/7baf764bf9702c92b5468fead72dab2454c587ea